### PR TITLE
Provide vagrantfile option

### PR DESCRIPTION
**Pull request by smk4664**
Added a vagrantfile and a copy of the selfmedicate script that runs minikube without a VM. The copy was created to allow continued use of the existing selfmedicate script.

### CODE REVIEW

**Comment by Mierdin** [comment]
Awesome! I'll take a look at this shortly.

**Comment by smk4664** [comment]
I found a problem with this, I am correcting now and will add another commit. The line endings got messed up when copied over the files.

**Comment by smk4664** [comment]
Evidently that is just a problem with my local copy, it looks like it is using the unix line endings on github.

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  # Base Ubuntu Box
+  config.vm.box = "bento/ubuntu-16.04"
+
+  config.vm.hostname = "antidote-#{antidote_config['version'].to_s.tr('.', '')}"
+
+  config.vm.define "Antidote #{antidote_config['version']}"
+
+  # Please see (https://github.com/cogitatio/vagrant-hostsupdater) for more information
+  if defined?(VagrantPlugins::HostsUpdater)
+     config.hostsupdater.aliases = ["antidote-local"]
```
Comment (Mierdin) [inline Vagrantfile:83]: I'm not familiar with how this is expected to work, but it didn't seem to update my local hosts file. I had to change the hosts file myself.

**Review by Mierdin — COMMENTED** [review]
This is a very impressive piece of work. For the most part, this just worked. So kudos there. I also appreciate that you created all new files for this, so we can continue to use the old model while we can direct folks to "try out" this new way of doing things. I imagine sometime in the near future we'll delete the "old" and make this the new normal. I'll build some marketing around this so folks know to try this and provide feedback.

Just a few relatively minor comments/questions above - the main thing is making sure the hosts file change gets taken care of automatically, and it didn't seem to happen for me. Otherwise folks will not be able to connect until they do this manually. Or maybe I did something wrong - now that I'm back in the office I'll give it another go.

Other than that, this is looking really good. Thanks for doing this.

File: Vagrantfile (hunk 1 above)
Comment (smk4664) [reply Vagrantfile:83]: I should have put that you have to have the plugin installed.  The details are in that github link.  The IP shouldn't rotate, so it is not too bad having to manually add.

**Comment by Mierdin** [comment]
Ah, for some reason I forgot you have to install Vagrant plugins manually. So looks like this is the new procedure:

```
vagrant plugin install vagrant-hostsupdater
vagrant up
```

Does that look right? I'm testing it again right now but I'm guessing that's what I did wrong.

Also, this brings up the topic of an addition to the docs that covers this. Ideally it would be a separate page, just like these are separate scripts. And in the future when we make this the primary way of doing things, we'd delete the old doc too. Is this something you are interested in taking on? I may have some time to do this in the next week or two but didn't want to do it if you were planning on adding the doc yourself. Up to you.

**Comment by smk4664** [comment]
Yes, that looks great. In Windows, the escalation wasn't being handled, so I followed their article for adding it once then it being set automatically every time.  

Yes, I need to find the docs and fork the repository.  I will work on that this week.  Antidote halt && Antidote up the second time don't work as expected, so I also be trying to fix that.

**Comment by Mierdin** [comment]
Awesome. For the docs change, https://antidoteproject.readthedocs.io/en/latest/contribute/docs.html should be helpful.

Sounds like you want to make some additional changes to this PR so I'll hold off on merging for now. Just give me a heads up when you feel it's good to go. Thanks again!

**Comment by smk4664** [comment]
This is ready now.

**Review by Mierdin — APPROVED** [review]
Excellent - looks good to me! Thanks again for doing this. And let me know when you get a chance to make a docs PR - once we have that I'll start putting the word out for folks to give this a spin.

**Pull request merged**